cmake/TestImpactFramework/LYTestImpactFramework.cmake

```diff
@@ -324,31 +324,64 @@ function(ly_test_impact_write_gem_target_enumeration_file GEM_TARGET_TEMPLATE_FI
      configure_file(${GEM_TARGET_TEMPLATE_FILE} ${mapping_path})
 endfunction()
 
+#! ly_extract_aliased_target_dependencies: recursively extracts the aliases of a target to retrieve the true de-aliased target.
+#
+# \arg:TARGET target to de-alias
+function(ly_extract_aliased_target_dependencies TARGET)
```

Comment:
This function should accept the name of a variable that can be used as an output parameter, to populate with the extracted alias target dependencies.
See the [ly_get_gem_load_dependencies](https://github.com/o3de/o3de/blob/development/cmake/SettingsRegistry.cmake#L72-L134) function as an example

Also you can rename this function to `o3de_extract_aliased_target_dependencies`?

Reply:
>This function should accept the name of a variable that can be used as an output parameter, to populate with the extracted alias target dependencies.
See the [ly_get_gem_load_dependencies](https://github.com/o3de/o3de/blob/development/cmake/SettingsRegistry.cmake#L72-L134) function as an example

Fixed 👍 

>Also you can rename this function to o3de_extract_aliased_target_dependencies?

As a separate PR I will, because I will rename all the TIAF cmake stuff accordingly (it's largely self contained). For this PR i will keep as-is purely for consistency 👍